packages/client/src/components/app/forms/BBReferenceField.svelte

```diff
@@ -1,9 +1,31 @@
 <script>
   import RelationshipField from "./RelationshipField.svelte"
+  import { sdk } from "@budibase/shared-core"
+
+  export let defaultValue
+
+  function updateUserIDs(value) {
+    if (Array.isArray(value)) {
+      return value.map(val => sdk.users.getGlobalUserID(val))
+    } else {
+      return sdk.users.getGlobalUserID(value)
+    }
+  }
+
+  function updateReferences(value) {
+    if (sdk.users.containsUserID(value)) {
+      return updateUserIDs(value)
+    }
+    return value
+  }
 </script>
 
 <RelationshipField
   {...$$props}
   datasourceType={"user"}
   primaryDisplay={"email"}
+  defaultValue={updateReferences(defaultValue)}
+  onChange={value => {
+    return updateReferences(value)
+  }}
```

Comment:
Actually wait - this `onChange` function won't do anything. This is currently called when you configure on click actions, so we still need to call the `onChange` prop passed in to the reference field.

```js
export let onChange

...
<RelationshipField
  ...
  onChange={({ value }) => onChange({ value: updateReferences(value) })}
  ...
/>
...
```

should do it I think.

Reply:
Although I don't think we need to worry about onChange at all do we? The values broadcast from the picker should always be the global IDs, since thats where the options are loaded from. I reckon the default value is maybe the only place we need to care about?

Reply:
Just for clarity too, `onChange` is specifically only the callback for any "on change" settings configured in the design section. It's not used to actually update the value of the field - that's done by `fieldApi.setValue(value)`. But like I said in my comment above, I don't think we need to worry about that since the values set by the picker will always be global IDs since that's where we load the options from.